packages/plugin-coinbase/src/types.ts

```diff
@@ -81,3 +81,61 @@ export type TradeTransaction = {
     errorCode: string | null;
     transactionUrl: string | null;
 };
+
+export interface TokenContractContent {
+    contractType: "ERC20" | "ERC721" | "ERC1155";
+    name: string;
+    symbol: string;
+    network: string;
+    baseURI?: string;
+    totalSupply?: number;
+}
+
+export const TokenContractSchema = z.object({
+    contractType: z.enum(["ERC20", "ERC721", "ERC1155"]).describe("The type of token contract to deploy"),
+    name: z.string().describe("The name of the token"),
+    symbol: z.string().describe("The symbol of the token"),
+    network: z.string().describe("The blockchain network to deploy on"),
+    baseURI: z.string().optional().describe("The base URI for token metadata (required for ERC721 and ERC1155)"),
+    totalSupply: z.number().optional().describe("The total supply of tokens (only for ERC20)"),
+}).refine(data => {
+    if (data.contractType === "ERC20") {
+        return typeof data.totalSupply === "number" || data.totalSupply === undefined;
+    }
+    if (["ERC721", "ERC1155"].includes(data.contractType)) {
+        return typeof data.baseURI === "string" || data.baseURI === undefined;
+    }
+    return true;
+}, {
+    message: "Invalid token contract content",
+    path: ["contractType"],
+});
+
+export const isTokenContractContent = (obj: any): obj is TokenContractContent => {
```

Comment:
not used anywhere

Reply:
Both are used in packages/plugin-coinbase/src/plugins/tokenContract.ts